scripts/utils.py

```diff
@@ -2,19 +2,176 @@
 Shared functions between various scripts
 """
 import re
+import sys
+
+try:
+    import tomllib
+except ModuleNotFoundError:
+    # pre-Python 3.11 compatibility
+    import toml as tomllib
```

Comment:
This will fail in existing CI bullseye pipelines because we don't activate the virtualenv: https://github.com/freedomofpress/securedrop-log/blob/487707a028b19931b013f0ea66d953fb1af20914/.circleci/config.yml#L32

So either we could go around and add `source .venv/bin/activate` to all the CI pipelines before merging this, or we could also catch this import, set tomllib to False and e.g. `print("tomllib/toml not found, poetry support unavailable")`

Reply:
I think I like the second solution better, will poke.

Reply:
OK, this is done minimally. As far as I can tell, pipelines should still succeed against packages that don't use Poetry. The `tomllib` name is only used inside the Poetry-related functions and there should be no `NameError` during a run against a non-poetry package.